runtime/frequency/src/lib.rs

```diff
@@ -304,6 +304,7 @@ pub type SignedExtra = (
 	pallet_frequency_tx_payment::ChargeFrqTransactionPayment<Runtime>,
 	pallet_msa::CheckFreeExtrinsicUse<Runtime>,
 	pallet_handles::handles_signed_extension::HandlesSignedExtension<Runtime>,
+	frame_metadata_hash_extension::CheckMetadataHash<Runtime>,
```

Comment:
Should we do this for passkey unsigned ext? Maybe yes

Reply:
This is to ensure that the offline wallets are signing the expected data structure (metadata). Since we are doing the signing via passkey adding this does not bring any additional value at this point since the metadata is a totally foreign concept from passkey's perspective.

We might be able to utilize this check in future by adding this metadata hash inside `PasskeyCall<T: Config>` type and that would allow us to ensure the passkey signature was done on a correct metadata.

Reply:
Yes, I’ve been considering it. Since this is an unsigned transaction, it might not be the best fit. However, I believe we are still signing the call function within the outer call. I wonder if we can verify that the hash of the metadata used at the time of signing matches the root.